charon/src/bin/charon/cli_rework/cargo.rs

```diff
@@ -10,3 +13,49 @@ pub struct CargoArgs {
     #[arg(last = true)]
     pub cargo: Vec<String>,
 }
+
+// Differences with current cargo call:
+// * cargo build rather than cargo rustc
+// * no longer set --lib and --bin: need to prune lib and bin fields in toml_config
+pub fn run_charon_cargo(mut options: CliOpts, host: &str) -> Result<ExitStatus> {
```

Comment:
I'd much rather you don't duplicate the logic. Switching to `cargo build` should work, and I don't see the harm in passing `--lib` ourselves if the user requests it.

Reply:
I'm making the `cargo build` change separately: https://github.com/AeneasVerif/charon/pull/607